reactivesocket-cpp/test/tcp/TcpClient.cpp

```diff
@@ -0,0 +1,112 @@
+#include "reactivesocket-cpp/src/tcp/TcpDuplexConnection.h"
+#include <thread>
+#include <folly/Memory.h>
+#include <reactivesocket-cpp/test/simple/PrintSubscriber.h>
+#include <reactivesocket-cpp/src/mixins/MemoryMixin.h>
+#include <reactivesocket-cpp/src/ReactiveSocket.h>
+#include <reactivesocket-cpp/src/RequestHandler.h>
+#include <reactivesocket-cpp/test/simple/CancelSubscriber.h>
+#include <reactivesocket-cpp/test/simple/CancelSubscription.h>
+#include <reactivesocket-cpp/src/framed/FramedDuplexConnection.h>
+
+using namespace ::testing;
+using namespace ::reactivesocket;
+using namespace ::folly;
+
+DEFINE_string(address, "[::]:9898", "host:port to connect to");
+
+namespace {
+    class ClientRequestHandler : public RequestHandler {
+    public:
+        ~ClientRequestHandler() {
+          LOG(INFO) << "~ClientRequestHandler()";
+        }
+
+        /// Handles a new Channel requested by the other end.
+        ///
+        /// Modelled after Producer::subscribe, hence must synchronously call
+        /// Subscriber::onSubscribe, and provide a valid Subscription.
+        Subscriber<Payload> &handleRequestChannel(
+            Payload request,
+            Subscriber<Payload> &response) override {
+          LOG(ERROR) << "not expecting server call";
+          response.onError(std::runtime_error("incoming request not supported"));
+
+          auto* subscription = new MemoryMixin<CancelSubscription>();
+          response.onSubscribe(*subscription);
+
+          return *(new MemoryMixin<CancelSubscriber>());
+        }
+
+        /// Handles a new inbound Subscription requested by the other end.
+        void handleRequestSubscription(Payload request, Subscriber<Payload> &response)
+        override {
+          LOG(ERROR) << "not expecting server call";
+          response.onError(std::runtime_error("incoming request not supported"));
+
+          auto* subscription = new MemoryMixin<CancelSubscription>();
+          response.onSubscribe(*subscription);
+        }
+    };
+
+    class Callback : public AsyncSocket::ConnectCallback {
+    public:
+        virtual ~Callback() = default;
+
+        void connectSuccess() noexcept override {
+        }
+
+        void connectErr(const AsyncSocketException &ex) noexcept override {
+          std::cout << "TODO error" << ex.what() << " " << ex.getType() << "\n";
+        }
+    };
+}
+
+int main(int argc, char *argv[]) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+  google::InitGoogleLogging(argv[0]);
+  google::InstallFailureSignalHandler();
+
+  EventBase eventBase;
+  auto thread = std::thread([&]() {
+      eventBase.loopForever();
+  });
+
+  std::unique_ptr<ReactiveSocket> reactiveSocket;
+  Callback callback;
+
+  eventBase.runInEventBaseThreadAndWait([&callback, &reactiveSocket, &eventBase]() {
+      folly::AsyncSocket::UniquePtr socket(new folly::AsyncSocket(&eventBase));
+
+      folly::SocketAddress addr;
+      addr.setFromLocalIpPort(FLAGS_address);
```

Comment:
if the address value will be "host:port", I think setFromLocalIpPort wont work.
(you can use the constructor of SocketAddress with a boolean flag to do the name resolution (localhost))